staging/src/k8s.io/client-go/discovery/aggregated_discovery.go

```diff
@@ -24,44 +24,72 @@ import (
 	"k8s.io/apimachinery/pkg/runtime/schema"
 )
 
+// StaleGroupVersionError encasulates failed GroupVersion marked "stale"
+// in the returned AggregatedDiscovery format.
+type StaleGroupVersionError struct {
```

Comment:
Why not reuse the existing struct? https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/discovery/discovery_client.go#L358

I'd imagine we may have clients who already cast to that type already and introducing a new type might be a bit confusing? Logically, stale is equivalent to discovery failed.

Reply:
The `ErrGroupDiscoveryFailed` error is supposed to encasulate **all** the Group/Versions which failed, which is why is stores a `map[schema.GroupVersion]error`. So it would be especially awkward (especially printing the error) if there were a hierarchy of these errors. BTW, the `ErrGroupDiscoveryFailed` error is what gets returned to the discovery interface user storing the individual errors in the map (as an example, see `ServerGroupsAndResources`).

Reply:
Thanks!